vms/platformvm/network/warp.go

```diff
@@ -0,0 +1,336 @@
+			Code:    common.ErrUndefined.Code,
+			Message: "failed to get subnet only validator: " + err.Error(),
+		}
+	}
+	return nil
+}
+
+// verifySubnetValidatorCanNotValidate verifies that the validationID is not
+// currently a validator.
+func (s signatureRequestVerifier) verifySubnetValidatorNotCurrentlyRegistered(
+	validationID ids.ID,
+	justification *platformvm.SubnetIDIndex,
+) *common.AppError {
+	subnetID, err := ids.ToID(justification.GetSubnetId())
+	if err != nil {
+		return &common.AppError{
+			Code:    ErrFailedToParseSubnetID,
+			Message: "failed to parse subnetID: " + err.Error(),
+		}
+	}
+
+	justificationID := subnetID.Append(justification.GetIndex())
+	if validationID != justificationID {
+		return &common.AppError{
+			Code:    ErrMismatchedValidationID,
+			Message: fmt.Sprintf("validationID %q != justificationID %q", validationID, justificationID),
+		}
+	}
+
+	s.stateLock.Lock()
+	defer s.stateLock.Unlock()
+
+	// Verify that the validator does not currently exist
+	_, err = s.state.GetSubnetOnlyValidator(validationID)
+	if err == nil {
+		return &common.AppError{
+			Code:    ErrValidationExists,
+			Message: fmt.Sprintf("validation %q exists", validationID),
+		}
+	}
+	if err != database.ErrNotFound {
+		return &common.AppError{
+			Code:    common.ErrUndefined.Code,
+			Message: "failed to lookup subnet only validator: " + err.Error(),
+		}
+	}
+	return nil
+}
+
+// verifySubnetValidatorCanNotValidate verifies that the validationID does not
+// currently and can never become a validator.
+func (s signatureRequestVerifier) verifySubnetValidatorCanNotValidate(
+	validationID ids.ID,
+	justificationBytes []byte,
+) *common.AppError {
+	justification, err := message.ParseRegisterSubnetValidator(justificationBytes)
+	if err != nil {
+		return &common.AppError{
+			Code:    ErrFailedToParseRegisterSubnetValidator,
+			Message: "failed to parse RegisterSubnetValidator justification: " + err.Error(),
+		}
+	}
+
+	justificationID := justification.ValidationID()
+	if validationID != justificationID {
+		return &common.AppError{
+			Code:    ErrMismatchedValidationID,
+			Message: fmt.Sprintf("validationID %q != justificationID %q", validationID, justificationID),
+		}
+	}
+
+	s.stateLock.Lock()
+	defer s.stateLock.Unlock()
+
+	// Verify that the validator does not and can never exists
+	_, err = s.state.GetSubnetOnlyValidator(validationID)
+	if err == nil {
+		return &common.AppError{
+			Code:    ErrValidationExists,
+			Message: fmt.Sprintf("validation %q exists", validationID),
+		}
+	}
+	if err != database.ErrNotFound {
+		return &common.AppError{
+			Code:    common.ErrUndefined.Code,
+			Message: "failed to lookup subnet only validator: " + err.Error(),
+		}
+	}
+
+	currentTimeUnix := uint64(s.state.GetTimestamp().Unix())
+	if justification.Expiry <= currentTimeUnix {
+		// The validator is not registered and the expiry time has passed
+		return nil
+	}
+
+	hasExpiry, err := s.state.HasExpiry(state.ExpiryEntry{
```

Comment:
```suggestion
    // If the validation ID was successfully registered and then removed, 
    // then it can never be re-used again even if its expiry has not yet passed.
	hasExpiry, err := s.state.HasExpiry(state.ExpiryEntry{
```